src/shared/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,81 @@
+// this rule gives trouble on forward refed components
+/* eslint-disable react/display-name */
+import React from 'react'
+import Icon from '../Icon'
+import { Checkmark, Container, InnerContainer, Input } from './Checkbox.styles'
+
+type HTMLCheckboxProps = React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>
+export interface CheckboxProps extends Omit<HTMLCheckboxProps, 'value' | 'onChange' | 'checked' | 'multiple' | 'ref'> {
+  value: boolean
+  disabled?: boolean
+  indeterminate?: boolean
+  error?: boolean
+  className?: string
+  onChange?: (value: boolean) => void
+  onFocus?: (event: React.FocusEvent<HTMLDivElement>) => void
+  onBlur?: (event: React.FocusEvent<HTMLDivElement>) => void
+}
+
+export const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
+  ({ value, indeterminate, onChange, disabled = false, onFocus, onBlur, error = false, className, ...props }, ref) => {
+    const isIndeterminate = indeterminate || false
+    const isSelected = !!value
+    const [isHovered, setIsHovered] = React.useState<boolean>(false)
+    const [isFocused, setIsFocused] = React.useState<boolean>(false)
```

Comment:
```suggestion
    const [isHovered, setIsHovered] = useState(false)
    const [isFocused, setIsFocused] = useState(false)
```
You don't need that.